api/canary/canary.go

```diff
@@ -25,51 +26,50 @@ func New(opts *Options) *Manager {
 	return &Manager{db: db}
 }
 
-func (cm *Manager) IsAdopter(username string) bool {
+func (cm *Manager) GetVersion(username string) string {
 	var adopter string
-	query := "select username,adopter from cbox_canary where username=?"
+	query := "select username, adopter from cbox_canary where username=?"
 	if err := cm.db.QueryRow(query, username).Scan(&username, &adopter); err != nil {
 		fmt.Println(err)
-		return false
+		return "production"
 	}
-	fmt.Println(adopter)
+
+	// For old data...
 	if adopter == "yes" {
-		return true
+		return "canary"
+	} else if adopter == "no" || adopter == "" {
+		return "production"
 	}
-	return false
+
+	return adopter
 }
 
-func (cm *Manager) SetStatus(username string, adopter bool) error {
-	isAdopter := "no"
-	if adopter {
-		isAdopter = "yes"
+func (cm *Manager) SetVersion(username, version string) error {
+
+	valid := Find([]string{"production", "canary", "ocis"}, version)
+	if !valid {
+		errors.New("Invalid option")
```

Comment:
return here and log the invalid option given

Reply:
ups..

Reply:
Fixed! the log happens in ocproxy.